include/revng/Model/Binary.h

```diff
@@ -118,6 +118,16 @@ class model::Binary : public model::generated::Binary {
 
   model::TypePath recordNewType(UpcastablePointer<Type> &&T);
 
+  /// \todo limit \tparam NewType using `SpecializationOf<model::Type>` concept
```

Comment:
Don't put TODO as doxygen

Reply:
Gotcha.

Reply:
This comment was not needed at all 😅.

`std::derived_from` is sufficient in this case, since `model::Type` is not a template.

Reply:
I'm closing this thread and will keep track of `std::derived_from` in the thread related to renaming this new member.